src/esi/Esi.cc

```diff
@@ -930,7 +930,7 @@ void
 ESIContext::addStackElement (ESIElement::Pointer element)
 {
     /* Put on the stack to allow skipping of 'invalid' markup */
-    Must(parserState.stackdepth < 10);
+    Must2(parserState.stackdepth < 10, "Too many nested elements");
```

Comment:
I do not insist on this change, but it would be more logical and clear to move this Must() to where we actually add the element to the stack using the being-checked `parserState.stackdepth` index. In its current location, this Must() incorrectly implies that we are protecting the `parserState.top()->addElement()` rather than `parserState.stack[parserState.stackdepth]` expression.

Reply:
resolved with the the exception model